.github/workflows/ci.yml

```diff
@@ -19,18 +18,76 @@ jobs:
     runs-on: ${{ matrix.os }}
     strategy:
       fail-fast: false
+
       matrix:
         os:
-          - windows-2019
           - ubuntu-20.04
           - macos-10.15
         compiler:
           # you can specify the version after `-` like "llvm-13.0.0".
           - llvm
           - gcc
+        generator:
+          - "Ninja Multi-Config"
+        build_type:
+          - Release
+          - Debug
+        developer_mode:
+          - ON
+          - OFF
+
         include:
-          - os: "windows-latest"
-            compiler: "msvc"
+          - os: ubuntu-20.04
+            compiler: gcc
+            generator: "Unix Makefiles"
+            build_type: Debug
+            developer_mode: On
+
+          - os: windows-2019
+            compiler: msvc
+            generator: "Visual Studio 16 2019"
+            build_type: Debug
+            developer_mode: On
+          - os: windows-2019
+            compiler: msvc
+            generator: "Visual Studio 16 2019"
+            build_type: Release
+            developer_mode: On
+          - os: windows-2019
+            compiler: msvc
+            generator: "Visual Studio 16 2019"
+            build_type: Debug
+            developer_mode: Off
+          - os: windows-2019
+            compiler: msvc
+            generator: "Visual Studio 16 2019"
+            build_type: Release
+            developer_mode: Off
+          - os: windows-2022
+            compiler: msvc
+            generator: "Visual Studio 17 2022"
+            build_type: Debug
+            developer_mode: On
+          - os: windows-2022
+            compiler: msvc
+            generator: "Visual Studio 17 2022"
+            build_type: Release
+            developer_mode: On
+          - os: windows-2022
+            compiler: msvc
+            generator: "Visual Studio 17 2022"
+            build_type: Debug
+            developer_mode: Off
+          - os: windows-2022
+            compiler: msvc
+            generator: "Visual Studio 17 2022"
+            build_type: Release
+            developer_mode: Off
```

Comment:
```suggestion
          - os: windows-2022
            compiler: msvc-2022
            generator: "Visual Studio 17 2022"
            build_type: Debug
            developer_mode: On
          - os: windows-2022
            compiler: msvc-2022
            generator: "Visual Studio 17 2022"
            build_type: Release
            developer_mode: On
          - os: windows-2022
            compiler: msvc-2022
            generator: "Visual Studio 17 2022"
            build_type: Debug
            developer_mode: Off
          - os: windows-2022
            compiler: msvc-2022
            generator: "Visual Studio 17 2022"
            build_type: Release
            developer_mode: Off
```

Reply:
The default version of msvc is [2019](https://github.com/aminya/setup-cpp/blob/61a3c6cd92576fce4081d1ecf15206c2d341936a/src/default_versions.ts#L2). I can probably add a logic that uses the GitHub Actions runner name to change this to 2022 if windows-2022 is used.